main.nf

```diff
@@ -902,11 +903,14 @@ process featureCounts {
     } else if (reverse_stranded && !unstranded){
         featureCounts_direction = 2
     }
+    // Try to get real sample name
+    sample_name = bam_featurecounts.baseName - ~/Aligned.sortedByCoord.out/
     """
     featureCounts -a $gtf -g gene_id -o ${bam_featurecounts.baseName}_gene.featureCounts.txt -p -s $featureCounts_direction $bam_featurecounts
     featureCounts -a $gtf -g gene_biotype -o ${bam_featurecounts.baseName}_biotype.featureCounts.txt -p -s $featureCounts_direction $bam_featurecounts
-    cut -f 1,7 ${bam_featurecounts.baseName}_biotype.featureCounts.txt | tail -n +3 > tmp_file
-    cat $biotypes_header tmp_file >> ${bam_featurecounts.baseName}_biotype_counts_mqc.txt
+    cut -f 1,7 ${bam_featurecounts.baseName}_biotype.featureCounts.txt | tail -n +3 | cat $biotypes_header - >> ${bam_featurecounts.baseName}_biotype_counts_mqc.txt
+    awk 'BEGIN{FS="\t"; print "Sample\tpercent_rRNA"}(\$0 !~ /^#/){total_count += \$2; if(\$1 == "rRNA"){rna_count=\$2}}END{print \"$sample_name\" FS (rna_count/total_count)*100;}' \
```

Comment:
🙇

Reply:
I know, either 'awk' one liner or a script in `bin` and I went with cool _(but eye-straining)_ way 😅

Reply:
Yeah, I have no idea what this is doing. As long as you know that if there's ever a bug that you have to fix it...!

But yeah, personally I'd generally prefer something that's longer (in `bin`) that can actually be understood and worked on given the choice 😉